test/functional/neutronless/loadbalancer/test_loadbalancer_opflex.py

```diff
@@ -0,0 +1,264 @@
+
+
+@pytest.fixture(scope="module")
+def icd_config():
+    oslo_config_filename = (
+        os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                     '../../config/basic_agent_config.json')
+    )
+    OSLO_CONFIGS = json.load(open(oslo_config_filename))
+
+    config = deepcopy(OSLO_CONFIGS)
+    config['icontrol_hostname'] = pytest.symbols.bigip_mgmt_ip_public
+    config['icontrol_username'] = pytest.symbols.bigip_username
+    config['icontrol_password'] = pytest.symbols.bigip_password
+    config['f5_vtep_selfip_name'] = pytest.symbols.f5_vtep_selfip_name
+
+    return config
+
+
+def test_create_delete_basic_lb(bigip, services, icd_config, icontrol_driver):
+
+    service_iter = iter(services)
+    service = service_iter.next()
+    lb_reader = LoadbalancerReader(service)
+    env_prefix = icd_config['environment_prefix']
+    fake_rpc = icontrol_driver.plugin_rpc
+    hostname = pytest.symbols.bigip_hostname
+
+    folder = '%s_%s' % (env_prefix, lb_reader.tenant_id())
+
+    # Make sure we are starting clean.
+    assert not bigip.folder_exists(folder)
+
+    # Create the loadbalancer
+    retval = icontrol_driver._common_service_handler(service)
+    assert retval
+
+    # Assert that update loadbalancer status was not called
+    assert fake_rpc.get_call_count('update_loadbalancer_status') == 0
+
+    # Assert folder created
+    assert bigip.folder_exists(folder)
+
+    # Assert route domain created
+    rd_name = folder
+    assert bigip.resource_exists(ResourceType.route_domain, rd_name)
+    rd = bigip.get_resource(ResourceType.route_domain,
+                            rd_name,
+                            partition=folder)
+
+    # Assert route domain properties
+    assert rd
+    assert rd.id == 1
+    assert rd.strict == "disabled"
+    # assert len(rd.vlans) == 0
+
+    # Assert that a self ip was created.
+    selfip_name = "local-%s-%s" % (hostname, lb_reader.subnet_id())
+    assert not bigip.resource_exists(ResourceType.selfip, selfip_name)
+
+    # Assert that a snat pool was created.
+    snatpool_name = folder
+    assert not bigip.resource_exists(ResourceType.snatpool, snatpool_name)
+
+    # Assert virtual address
+    virtual_addr_name = "%s_%s" % (env_prefix, lb_reader.id())
+    assert not bigip.resource_exists(ResourceType.virtual_address,
+                                     virtual_addr_name)
+
+    # Create the VLAN
+    service = service_iter.next()
+    icontrol_driver._common_service_handler(service, delete_partition=True)
+    lb_reader = LoadbalancerReader(service)
+
+    # Assert tunnel created.
+    vlan_name = 'vlan-%d' % (lb_reader.network_seg_id())
+    fq_vlan_name = '/%s/%s' % (folder, vlan_name)
+    assert bigip.resource_exists(ResourceType.vlan, vlan_name)
+    vlan = bigip.get_resource(ResourceType.vlan,
+                              vlan_name,
+                              partition=folder)
+    assert vlan
+
+
+    # Assert route domain created
```

Comment:
Can this entire test file and the previous one be converted to a data-driven table of feeding in json files and declaring expected outcomes?